discovery/proxy.go

```diff
@@ -36,23 +40,50 @@ func ProxyToRendezvous(original Discovery, servers []ma.Multiaddr, topic string,
 			wg.Wait()
 			return nil
 		case <-lookup:
+		case n := <-closeRequest:
+			if _, exist := lastSeen[n]; !exist {
+				continue
+			}
+			// closeRequest is sent every time window after record was seen.
+			// record must be discovered again during same time window otherwise it will be removed.
+			if time.Since(lastSeen[n]) >= timeWindow {
+				close(closers[n])
+				_ = identities[n].Stop()
+				delete(identities, n)
+				delete(lastSeen, n)
+				delete(closers, n)
+				total--
+				log.Info("proxy for a record was removed", "identity", n.String(), "total", total)
+			}
 		case n := <-found:
-			if _, exist := identities[n.ID]; exist {
+			_, exist := identities[n.ID]
+			// skip new record if we reached a limit.
+			if !exist && total == limit {
+				continue
+			}
+			lastSeen[n.ID] = time.Now()
+			go time.AfterFunc(timeWindow, func() {
```

Comment:
I think a better solution would be to have a separate timer triggering a loop that runs every 1 second and checks last seen time of all identities because:
* we need to be careful to not close `closeRequest` channel while timers are still scheduled,
* we limit by the number of found nodes but not the number of how many founds happen. This can lead to issuing thousands of goroutines (shouldn't be a problem but e.g. debugging might be hard in case of a panic).

Also, it looks like `time.AfterFunc` does not block and executes its function argument in a goroutine.